examples/main.rs

```diff
@@ -22,8 +22,8 @@ struct Data {
 }
 
 async fn post_data(
-    Json(payload): Json<Data>,
     Extension(state): Extension<Arc<Mutex<State>>>,
+    Json(payload): Json<Data>,
```

Comment:
Is there a reason for this change?

Reply:
I too find it a bit odd, but changing the position of the argument fixes the compile error.

![image](https://user-images.githubusercontent.com/47286750/205074649-c76ecf29-3bef-43f0-91c3-a8e951c7e03e.png)

Reply:
Huh - yeah, I just went through and upgraded a couple projects to axum 0.6 today and noticed this as well. Thanks!